client/state/data-layer/wpcom/sites/admin-menu/index.js

```diff
@@ -25,6 +26,16 @@ const sanitizeUrl = ( url, wpAdminUrl ) => {
 		url?.replace( /^https?:\/\//, '' )
 	);
 
+	if ( isSafeWpAdminUrl ) {
+		url = addQueryArgs(
+			{
+				return: document.location.href, // Gives WP Admin a chance to return to where we started from.
+				from: 'calypso-unified-menu', // `from` param is used by WP Admin on Atomic sites for enabling Nav Unification in that context. Can be removed after rolling Nav Unification out to 100% of users.
```

Comment:
Could we explore adding `from: 'calypso-unified-menu'` this only for Atomic sites? I am not tied to this, just dropping an idea.

Reply:
I tried it, but suddenly some tests start to fail because the state is not properly configured on the existing tests, so I'm going to leave it as-is. Given this is only a temporary solution, I think it's fine if it doesn't target Atomic sites exclusively.

Reply:
Good point, I always forget this is temporary (or at least we hope so 😁)!